test/widget/nav_bar/data/nav_bar_item_test.dart

```diff
@@ -0,0 +1,195 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+void main() {
+  group(
+    'icon button',
+    () {
+      const iconPath = 'iconPath';
+      const isHighlighted = true;
+      const key = Key('iconBtnKey');
+      void onPressed() {}
+
+      final item = NavBarItemIconButton(
+        svgIconPath: iconPath,
+        isHighlighted: isHighlighted,
+        onPressed: onPressed,
+        key: key,
+      );
+
+      test(
+        'GIVEN item THEN verify values are equal',
+        () {
+          expect(item.svgIconPath, iconPath);
+          expect(item.onPressed, onPressed);
+          expect(item.isHighlighted, isHighlighted);
+          expect(item.key, key);
+        },
+      );
+
+      test(
+        'GIVEN item THEN verify it extends NavBarItem and Equatable ',
+        () {
+          expect(item, isA<NavBarItem>());
+          expect(item, isA<Equatable>());
+        },
+      );
+
+      test(
+        'GIVEN item THEN verify correct props',
+        () {
+          expect(item.props, [
+            key,
+            isHighlighted,
+            iconPath,
+          ]);
+        },
+      );
+    },
+  );
+  group(
+    'back button',
+    () {
+      const key = Key('backBtnKey');
+      void onPressed() {}
+
+      final item = NavBarItemBackButton(
+        onPressed: onPressed,
+        key: key,
+      );
+
+      test(
+        'GIVEN item THEN verify values are equal',
+        () {
+          expect(item.onPressed, onPressed);
+          expect(item.key, key);
+        },
```

Comment:
Not sure if these tests like these are really useful as they seem to test dart instead of testing functionality. This test for instance tests if you created a class and passed param to the constructor, the data members are still the same as you the params passed. wdyt?

Reply:
I added it, cos technically I can `wrap` voidCallback into another callback, which I want to avoid.
The same with the key. There is no guarantee that passed into constructor `key` will be the same as the variable.
```dart
  const NavBarItemBackButton({
    required VoidCallback onPressed,
    required Key key,
  }) : super._(false, (){ }, Key('another_key'));
```